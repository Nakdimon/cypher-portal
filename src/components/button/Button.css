.button_cont {
    margin: 30px 0px 0px;
}

.cypher-button {
    color: #fff !important;
    font-size: 25px;
    text-transform: uppercase;
    text-decoration: none;
    background: #213764;
    padding: 10px;
    border-radius: 5px;
    display: inline-block;
    border: none;
    transition: all 0.4s ease 0s;
}