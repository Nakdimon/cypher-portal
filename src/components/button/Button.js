import React from 'react';
import './Button.css'

function Button({
    isRequestButtonDisabled,
    handleCypherAction
}) {
    return (
        <div className="button_cont" align="center"><a className={'cypher-button'+' '+isRequestButtonDisabled} onClick={handleCypherAction}>Cypher Text!</a></div>
    )

}

export default Button;