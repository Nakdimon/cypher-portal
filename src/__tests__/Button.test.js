import React from 'react';
import { create } from 'react-test-renderer';
import Button from '../components/button/Button';

const handleCypherAction = () => 'dummyHandleCypherAction'

describe("Button component: ", ()=> {
  it("Matches the snapshot", () => {
    const button = create(
      <Button
        isRequestButtonDisabled={false}
        handleCypherAction={handleCypherAction}
      />
    )
    expect(button).toMatchSnapshot();
  })
})