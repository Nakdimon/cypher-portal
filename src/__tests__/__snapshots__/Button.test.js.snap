// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component:  Matches the snapshot 1`] = `
<div
  align="center"
  className="button_cont"
>
  <a
    className="cypher-button false"
    onClick={[Function]}
  >
    Cypher Text!
  </a>
</div>
`;

exports[`Button component:  Matches the snapshot 1`] = `
<div
  align="center"
  className="button_cont"
>
  <a
    className="cypher-button false"
    onClick={[Function]}
  >
    Cypher Text!
  </a>
</div>
`;
